packages/dds/tree/src/feature-libraries/modular-schema/modularChangeEncoding.ts

```diff
@@ -173,5 +179,9 @@ function decodeNodeChangesetFromJson(
 		);
 	}
 
+	if (encodedChange.valueConstraint !== undefined) {
+		decodedChange.valueConstraint = { value: encodedChange.valueConstraint, violated: false };
```

Comment:
Is there a reason we don't have to record whether the constraint was violated when serializing?